fileHub/js-app/app/services/state-management/state-manager.js

```diff
@@ -0,0 +1,77 @@
+import {mutator} from './mutator/mutator.js';
+
+/**
+ * Manages the states of the application by processing actions.
+ */
+export class StateManager {
+  /**
+   * @constructor
+   * @param {object} initialState
+   * @param {object} services
+   * @param {ActionFactory} actionFactory
+   */
+  constructor(initialState, services, actionFactory) {
+    this._initialState = initialState;
+    this._services = services;
+
+    this._eventBus = new EventTarget();
+    this._actions = actionFactory;
+
+    const eventBus = this._eventBus;
+    this._state = new Proxy(this._eventBus, {
+      set(target, key, newValue) {
+        if (target[key] === newValue) {
+          return true;
+        }
+
+        const updateTargetValue = Reflect.set(target, key, newValue);
+        if (updateTargetValue) {
+          eventBus.dispatchEvent(new CustomEvent(`stateChanged-${key}`, {
+            detail: {state: target},
+          }));
+          return true;
+        }
+      },
+    });
+  }
+
+  /**
+   * Changes application components states.
+   * @param {string} mutatorName
+   * @param {object} details
+   * @private
+   */
+  _mutate(mutatorName, details) {
+    const state = this._state;
+    const mutatedState = mutator(mutatorName, details, state);
+    this._state = Object.assign(this._state, mutatedState);
```

Comment:
Please refactor according to what was discussed (Proxy is used for notifying subscribers for state).